### DISCUSSION TITLE
Can't connect

### DISCUSSION THREAD

**Discussion by JimO67** [aqualinkd/AqualinkD#208] [Q&A]
Installing aqualinkd pi zw.  Got to part sudo aqualinkd - d -c /etc/aqualinkd.conf
That seems to work fine.  Long list of "Notice:" with only two errors about serial com at bottom of Notices.  I assume because I am not connected to the pool controller/RS485 port.

I determined the IP address and when I try to access via chrome on my laptop connected to he same router I get refusal to connect.

How do I trouble shoot this?

**Comment by ballle98** [comment]
Can you attach your aqualinkd.conf file or capture the logs and attach those?  If you are using a port other than :80 you will have to specify that in chrome.  I have also run into issues where apache got installed, but if that was the issue you would see an apache default page.  If apache is enabled you disable it "systemctl disable apache2".

**Reply by JimO67** [reply]
Thanks so much for the reply.  Port is 80.  I didn't install Apache.  New Pi Zero W.  I see the log file path is /var/www/aqualinkd   I found that folder but don't see any log files there.  How do you get files out of the pi and sent to my PC?  I can't run Chromium on the Pi zero.  Tried Links 2, its like going back to 1990.

**Reply by JimO67** [reply]
oh wait, log files go to /var/log but there is no aqualink log file there.  Oh duh.  The log file path is commented.  I have to remove # to get the log file to write.

**Reply by sfeakes** [reply]
Your config file has the settings for how and where to log, or just run without -d option to get log to stdout.
As for moving the files, use ssh or a ftp tool over ssh.

**Reply by JimO67** [reply]
I don't have much linux experience.  Do I need to kill the process that is running or just reissue the sudo aqualink ...  seems to rerun it.  
I don't seem to be getting a log file to be written,.  Took of the -d option, is stdout a folder and where would it be?

**Reply by sfeakes** [reply]
I’m sorry, long week.  Leave the -d in, (I had that backwards).  Stdout is the console / screen you are using unless you have something redirected.
So as you start, look through the logs that are printed out, you should see a message about starting a web server.  But since you have no RS control panel connected, you will need to wait for it to time out.   You may also want to edit the configuration file and setup logging to a file, and change the default panel I’d from Ox00 to something else as should make the rs connection time out quicker.
This is the message you are looking for.
`NetService:Starting web server on port 80`

**Reply by JimO67** [reply]
OK.  In the mean time I got FTP to work.  Its been decade + since I used this stuff.

**Reply by JimO67** [reply]
Ok, see log file and conf file.  I had to rename the conf file to txt to copy here.

[aqualinkd.log](https://github.com/sfeakes/AqualinkD/files/11392689/aqualinkd.log)
[aqualinkd.txt](https://github.com/sfeakes/AqualinkD/files/11392692/aqualinkd.txt)

**Reply by JimO67** [reply]
I figured out how to make a network location for my Raspberry Pi FTP File folder.
![FTP](https://user-images.githubusercontent.com/90992628/236109086-7073ac57-f494-4d57-830e-6f7fef039fc7.jpg)

https://cybertext.wordpress.com/2009/11/25/vista-create-ftp-shortcut-with-login-details/

**Reply by sfeakes** [reply]
From your config file, it looks like you have set the log level to warning, so you are not seeing all the information.  You only set log level once in the config, and since warning is the last entry, that’s what’s set.   Comment out all the LOG_LEVEL lines except the DEBUG one.  Then you will see more information in the log.

**Reply by JimO67** [reply]
I'll try that this afternoon.  I am at the house with the pool now.  Looks like I have an RS6.  See attached pics. There is a chip that says RS6/PDA Combo.  Front panel indicated RS6 or RS8.  I have two RS485 ports so I assume I can use the and AqualinkD.
I'll modify the config file at my desk and see what it tells me.  Also I will select RS-6.   Then I will  take it out to connect to the pool and see how that goes.  I can telnet out to that Pi and edit from the comfort of my office right?  Need to set up that.  I think that is how
![Screenshot 2023-05-04 135910](https://user-images.githubusercontent.com/90992628/236311285-65532c1f-d2b5-475c-8e25-b43dc69438f4.jpg)
 I worked on my Jetson Nano a few years ago when I was fascinated by neural networks.  Used Putty.  
![Screenshot 2023-05-04 140005](https://user-images.githubusercontent.com/90992628/236311305-bd24c7ee-e75e-4a74-8db7-023b2baffde3.jpg)
![Screenshot 2023-05-04 140109](https://user-images.githubusercontent.com/90992628/236311315-4e6a1309-7f6e-479d-9170-dbf20feadc67.jpg)

![Screenshot 2023-05-04 140137](https://user-images.githubusercontent.com/90992628/236311337-d73ef9e0-453c-4bfd-969f-91b495bd0e8c.jpg)

**Reply by JimO67** [reply]
Not seeing "Starting web server"

[aqualinkd.log](https://github.com/sfeakes/AqualinkD/files/11401777/aqualinkd.log)

Well, I plugged in the RS485 USB adapter.  Not connected to the RS-6 yet though.  Now I do get an error relevant to port 80.  

[RS485 Adapter -aqualinkd.log](https://github.com/sfeakes/AqualinkD/files/11402000/RS485.Adapter.-aqualinkd.log)

However, I can connect from web browser now.

![Screenshot 2023-05-04 172121](https://user-images.githubusercontent.com/90992628/236343688-c7c3f9df-a793-43a4-841c-042bbe6cdb90.jpg)
 I looked at the log again and it  is seeing my button presses and responding with log entries.  I think it's working.  

Now I need to understand setting up the config file with I have some confusion on what really needs edited or not.  The parts about device id's.  It seems Aqualinkd will read what it needs but then there options to enter ID's.  Using the panel pics I sent could you send a suitable config file?  I have single speed pumps.  One for filter and one for the waterfall (sheer).  I have a heater. I have a color light in the pool an spa but they are controlled together.   I have one PDA.  I can leave the transceiver on the first RS485 port and attach the pi zero to the 2nd and still use the PDA right?

Finally, is there any way I can screw up the settings in the aqualink controller that can't easily be recovered from?  If I have to do some trial and error.  All could be restored by using the PDA?

**Comment by sfeakes** [comment]
Along with the above, also check the logs to see if the web server has started.  If there is a port conflict, you should see that error. But my guess is it's still waiting for the control panel connection, so it hasn't even got to starting the web server yet.

**Reply by JimO67** [reply]
If it has to get past the control panel connection I guess I would have to wait until I get to the other house with the pool.  I assumed you could connect/bench test without the panel connected. Is there an easy way to edit some code and skip that part?  Are you the creator?
